addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1125,14 +1153,15 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
         this._onUserValueReset(ev);
     },
     /**
-     * Called when the color reset button is clicked -> removes all color
-     * classes and color styles.
-     *
-     * @private
+     * @override
      */
-    _onColorReset: function (ev) {
-        this._value = '';
-        this._onUserValueChange(ev);
+    _onClick: function (ev) {
+        // Do not close the colorpalette on colorpicker click
+        if (ev.target.closest('.o_colorpicker_widget')) {
+            ev.stopPropagation();
+            return;
+        }
```

Comment:
Mmh. Still not sure what I prefer to handle this :) I kinda like flagging the event in the widget itself and checking the flag here. It has the advantage to:
- not force stopPropagation in the widget itself
- allow the parent to stopPropagation of events of internal widgets without knowing their internal structure/main class

But this is fine by me too. I let you judge.

Reply:
I think you are right. When you say flagging the event you mean passing additional data to it in the colorpickerWidget ? Is it possible without customEvent ?

Reply:
An event is just an object, you can just add a property to it.

Widget:
```js
myHandler: function (ev) {
    ev.__thisEventWasSeen = true;
},
```
Parent:
```js
if (ev.__thisEventWasSeen) {
    ev.stopPropagation();
}
```
I would name the flag with the name of the widget class or something like that

Reply:
(I thought about making that generic to all widgets at some point: add an automatic handler to all widget which flags any event with the name of the widget... but the case is not that common anyway so too crappy to have code running for that).

Reply:
I know that but it looks like you can not alter an event that is bubbling. It looks like it is recreated each time it bubble. If it is captured on the same element no problem but on a parent the modifications are not replicated.

Reply:
Looks like it is working inside of the same widget but not from a wiget to another

Reply:
You are probably using the jQuery event object instead of the JavaScript event

Reply:
I have to do this then

Widget:

```js
myHandler: function (ev) {
    ev.originalEvent.__thisEventWasSeen = true;
},
```
Parent:

```js
if (ev.originalEvent.__thisEventWasSeen) {
    ev.stopPropagation();
}
```

Reply:
:+1: